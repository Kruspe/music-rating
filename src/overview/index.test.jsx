import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { API, Auth } from 'aws-amplify';
import MaterialTable from 'material-table';
import Overview from './index';

const expectedRatings = [{
  band: 'band', festival: 'Wacken', year: 2014, rating: 5,
}];

describe('Overview', () => {
  let getBandsSpy;
  beforeEach(() => {
    getBandsSpy = jest.spyOn(API, 'get').mockImplementation(() => expectedRatings);
    getBandsSpy.mockClear();
    const currentSessionMock = Promise.resolve({
      getAccessToken: () => ({ getJwtToken: () => ('Token') }),
    });
    const currentUserInfoMock = Promise.resolve({ id: 'userId' });
    jest.spyOn(Auth, 'currentUserInfo').mockImplementation(() => currentUserInfoMock);
    jest.spyOn(Auth, 'currentSession').mockImplementation(() => currentSessionMock);
  });
  it('should render correctly', (done) => {
    const wrapper = shallow(<Overview />);
    process.nextTick(() => {
      expect(toJSON(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should render rating correctly', (done) => {
    const wrapper = shallow(<Overview />);
    process.nextTick(() => {
      const ratingComponent = shallow(wrapper.find(MaterialTable).prop('columns').find((column) => column.field === 'rating').render(expectedRatings[0]));
      expect(ratingComponent.prop('name')).toEqual('band');
      expect(ratingComponent.prop('value')).toEqual(5);
      expect(ratingComponent.prop('readOnly')).toBeTruthy();
      done();
    });
  });

  it('should get user data', async () => {
    const wrapper = shallow(<Overview />);

    process.nextTick(() => {
      expect(getBandsSpy).toHaveBeenCalledWith('musicrating', '/bands/userId', { header: { Authorization: 'Bearer Token' } });
      expect(wrapper.state().ratings).toEqual(expectedRatings);
    });
  });
});
