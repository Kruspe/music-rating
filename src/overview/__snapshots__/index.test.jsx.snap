// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should render correctly 1`] = `
<Fragment>
  <WithStyles(Component)
    columns={
      Array [
        Object {
          "field": "band",
          "title": "Band",
        },
        Object {
          "field": "festival",
          "title": "Festival",
        },
        Object {
          "field": "year",
          "title": "Year",
        },
        Object {
          "field": "rating",
          "render": [Function],
          "title": "Rating",
        },
      ]
    }
    data={
      Array [
        Object {
          "band": "band",
          "festival": "Wacken",
          "rating": 5,
          "year": 2014,
        },
      ]
    }
    icons={
      Object {
        "FirstPage": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "LastPage": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "NextPage": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "PreviousPage": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "ResetSearch": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "Search": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "SortArrow": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    options={
      Object {
        "draggable": false,
        "pageSize": 10,
        "showTitle": false,
      }
    }
  />
</Fragment>
`;
